code/game/objects/items/weapons/grenades/stinger.dm

```diff
@@ -0,0 +1,62 @@
+/proc/stinger(var/source,var/fragx,var/fragy,var/light_dam,var/flash_dam,var/p_dam,var/p_range,var/can_cover=TRUE,var/shard_range = 50)
+	var/turf/O = get_turf(source)
+	var/fragger = rand(fragx,fragy)
+	explosion(O, -1, -1, light_dam, flash_dam)
+	var/list/target_turfs = getcircle(O, 7)
+	var/fragments_per_projectile = round(fragger/target_turfs.len)
+
+	for(var/turf/T in target_turfs)
+		sleep(0)
```

Comment:
Sleeps should be avoided, you can schedule callbacks from the timer subsystems to another function that handles what you do below, or simply remove the sleep and call this function with `INVOKE_ASYNC`